### PR TITLE
Add Windows support, including openssh and pageant agents; closes #19

### LINKED ISSUE #19: Windows 10 error: undefined: syscall.Stat_t
system: windows 10
golang 1.7.4

# github.com/reconquest/orgalorg
D:\Home\Go\src\github.com\reconquest\orgalorg\archive.go:190: undefined: syscall.Stat_t
D:\Home\Go\src\github.com\reconquest\orgalorg\archive.go:197: undefined: syscall.Stat_t
D:\Home\Go\src\github.com\reconquest\orgalorg\archive.go:201: undefined: syscall.Stat_t

### PR DESCRIPTION
**Pull request by cr1cr1**

### CODE REVIEW

**Comment by cr1cr1** [comment]
pinging @kovetskiy @seletskiy

File: main.go [hunk 1]
```diff
@@ -167,7 +166,9 @@ Advanced options:
                            locking and execution commands.
                            [default: 16].
   --no-preserve-uid       Do not preserve UIDs for transferred files.
```
Comment (kovetskiy) [inline main.go:168]: We could also export this to files with _windows.go and _unix.go suffix. What do you think? Is it necessary?

File: sshagent_windows.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	"github.com/Microsoft/go-winio"
+	"golang.org/x/crypto/ssh/agent"
+)
+
+const (
+	openSshAgentPipe = `\\.\pipe\openssh-ssh-agent`
+)
+
+func getSshAgent() (agent.Agent, error) {
+	pageant_window := pageant.Available()
```
Comment (kovetskiy) [inline sshagent_windows.go:18]: The use of snake_case is not idiomatic in Go and it's highly recommended to use camelCase instead.

In this case you could even avoid creating a variable and just check it like `if pageant.Available() { ... } else { ... }`

File: README.md [hunk 3]
```diff
@@ -29,7 +29,7 @@
 * Both strict or loose modes of failover to be sure that everything will either
   fail on any error or try to complete, no matter of what.
 
-* Interactive password authentication as well as SSH public key authentication.
+* Interactive password authentication as well as SSH public key authentication. Will use ssh-agent if present. On Windows, orgalorg can connect to **pageant** or **openssh agent**.
```
Comment (kovetskiy) [inline README.md:32]: Also, could you wrap the line to 80 symbols since the entire README.md file does it already?

**Review by kovetskiy — APPROVED** [review]
Overall, it looks good to me. A few nitpicks and we're good to go.

I'll be glad to merge such a feature.

File: main.go (hunk 1 above)
Comment (cr1cr1) [reply main.go:168]: Oh, yeah, totally right there :)

File: sshagent_windows.go (hunk 2 above)
Comment (cr1cr1) [reply sshagent_windows.go]: Ah, right.. that was a copy/paste.

**Comment by kovetskiy** [comment]
Thank you!

**Pull request merged**